src/iris/bin/sync_targets.py

```diff
@@ -336,11 +423,15 @@ def get_ldap_list_membership(l, search_strings, list_name):
                              ldap.SCOPE_SUBTREE,
                              serverctrls=[req_ctrl],
                              attrlist=[search_strings['user_mail_field']],
-                             filterstr=search_strings['user_membership_filter'] % escape_filter_chars(list_name.decode('utf-8'))
+                             filterstr=search_strings['user_membership_filter'] % escape_filter_chars(list_name)
                              )
         rtype, rdata, rmsgid, serverctrls = l.result3(msgid, timeout=ldap_timeout, resp_ctrl_classes=known_ldap_resp_ctrls)
 
-        results |= {data[1].get(search_strings['user_mail_field'], [None])[0] for data in rdata}
+        for data in rdata:
+            member = data[1].get(search_strings['user_mail_field'], [None])[0]
```

Comment:
From this it looks like if the user_mail_field field does not exist, we'll get back a None, and end up inserting a `None` into our results set. Do we want that?